src/greeter/GreeterApp.cpp

```diff
@@ -138,14 +138,14 @@ namespace SDDM {
         //view->setGeometry(QRect(QPoint(0, 0), screen->geometry().size()));
         view->setGeometry(screen->geometry());
         m_views.append(view);
-
+        m_view_mapping.insert(screen, view);
         // remove the view when the screen is removed, but we
         // need to be careful here since Qt will move the view to
         // another screen before this signal is emitted so we
         // pass a pointer to the view to our slot
-        connect(qGuiApp, &QGuiApplication::screenRemoved, this, [view, this](QScreen *) {
-            removeViewForScreen(view);
-        });
+        //connect(qGuiApp, &QGuiApplication::screenRemoved, this, [view, this](QScreen *) {
```

Comment:
Heh, this old code was very stupid, it doesn't check which view to remove.
Just adding an if() here would probably have sufficed.

I bet it used to be connect(screen, &QObject::destroyed,...) and someone did a terrible port.

In any case please don't leave commented out obsolete code.